aspnet/mvc/controllers/filters/sample/src/FiltersSample/Filters/DurationActionFilter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Diagnostics;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace FiltersSample.Filters
+{
+    public class DurationActionFilter : IActionFilter
+    {
+        private ILogger _logger;
+        private Stopwatch _stopwatch;
+        public DurationActionFilter(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<DurationActionFilter>();
+        }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            _stopwatch = Stopwatch.StartNew();
+        }
+
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+            var actionDuration = _stopwatch.ElapsedMilliseconds;
+            _stopwatch.Stop();
+            _logger.LogInformation($"{context.ActionDescriptor.DisplayName}: {actionDuration}ms");
+
+            // add time to viewbag if viewresult
+            ViewResult result = context.Result as ViewResult;
+            if (result != null)
+            {
+                result.ViewData["actionDuration"] = actionDuration;
+            }
```

Comment:
@rynowak How about steve say a real app would use Glimpse for profiling. This timing filter is just for demo and it's a good simple sample that actually does something useful. Customers loved playing with my stopwatch timer even tho I told them to read my MVC/Glimpse article for real perf work. 
blowdart won't allow bad security with a caveat "don't do what I'm doing in this sample" - but in this case is it OK to say "Don't use this for real perf work, use Glimpse"